File: python/03_catalogue_and_models.py
"""Public catalogue + public models (no key) and runtime models (with key).

    export TONIA_API_KEY=tonia_sk_…   # only needed for runtime models
    python 03_catalogue_and_models.py
"""

from __future__ import annotations

import os

from tonia import Tonia

with Tonia() as public_client:
    catalogue = public_client.catalogue.list()
    public_models = public_client.public_models.list()
    print("catalogue products:", len(catalogue.get("products", [])))
    print("public models:", len(public_models.get("data", [])))

with Tonia(api_key=os.environ.get("TONIA_API_KEY")) as client:
    runtime = client.models.list()
    print("runtime models:", len(runtime.get("data", [])))
    print("last_limits:", client.last_limits)

File: typescript/03-catalogue-and-models.ts
/**
 * Public catalogue + public models (no key) and runtime models (with key).
 *
 *   export TONIA_API_KEY=tonia_sk_…   # only needed for runtime models
 *   npx tsx 03-catalogue-and-models.ts
 */
import { Tonia } from "@tonia-router/sdk";

const publicClient = new Tonia(); // no key for public routes
const catalogue = await publicClient.catalogue.list();
const publicModels = await publicClient.publicModels.list();
console.log("catalogue products:", (catalogue as { products?: unknown[] }).products?.length);
console.log("public models:", (publicModels as { data?: unknown[] }).data?.length);

const client = new Tonia({ apiKey: process.env.TONIA_API_KEY });
const runtime = await client.models.list();
console.log("runtime models:", (runtime as { data?: unknown[] }).data?.length);
// Bearer / OpenAI-shaped ids. x-api-key listing looks different.
console.log("lastLimits:", client.lastLimits);
